### PR TITLE
Hw 03 third part sql

### PR DESCRIPTION
**Pull request by Alex-Korzun**
Added Travis, but it will work after merging I think.

### CODE REVIEW

File: src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+@Dao
+public class ManufacturerDaoJdbcImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String query = "INSERT INTO manufacturers "
+                + "(manufacturer_name, manufacturer_country) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query,
+                    Statement.RETURN_GENERATED_KEYS);
```
Comment (vlykhvar) [inline src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java]:
```suggestion
        try (Connection connection = ConnectionUtil.getConnection();
            PreparedStatement preparedStatement = connection.prepareStatement(query,
                    Statement.RETURN_GENERATED_KEYS)) {
```

File: src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            preparedStatement.setString(1, manufacturer.getName());
+            preparedStatement.setString(2, manufacturer.getCountry());
+            preparedStatement.executeUpdate();
+
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                manufacturer.setId(resultSet.getObject(1, Long.class));
+            }
+            preparedStatement.close();
```
Comment (vlykhvar) [inline src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java]:
```suggestion
            
```

File: src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> getById(Long manufacturerId) {
+        String query = "SELECT * FROM manufacturers "
+                + "WHERE manufacturer_id = ? AND `delete` = FALSE";
+        Manufacturer manufacturer = null;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
```
Comment (vlykhvar) [inline src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java]:
```suggestion
        try (Connection connection = ConnectionUtil.getConnection();
            PreparedStatement preparedStatement = connection.prepareStatement(query)) {
```

File: src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+            preparedStatement.setLong(1, manufacturerId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                manufacturer = createManufacturer(resultSet);
+            }
+            preparedStatement.close();
```
Comment (vlykhvar) [inline src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java]:
```suggestion
           
```

**Review by vlykhvar — APPROVED** [review]

File: src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java (hunk 1 above)
Comment (Alex-Korzun) [reply src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java]: done

File: src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java (hunk 2 above)
Comment (Alex-Korzun) [reply src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java]: done

File: src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java (hunk 3 above)
Comment (Alex-Korzun) [reply src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java]: done

File: src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java (hunk 4 above)
Comment (Alex-Korzun) [reply src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java]: done

File: src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+            throw new DataProcessingException("Can't create manufacturer "
+                    + manufacturer, e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> getById(Long manufacturerId) {
+        String query = "SELECT * FROM manufacturers "
+                + "WHERE manufacturer_id = ? AND `delete` = FALSE";
```
Comment (RomanDubovskyi) [inline src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java]:
```suggestion
                + "WHERE manufacturer_id = ? AND deleted = FALSE";
```

File: src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+                + "WHERE `delete` = FALSE";
+        List<Manufacturer> manufacturers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+            PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                manufacturers.add(createManufacturer(resultSet));
+            }
+            preparedStatement.close();
+            return manufacturers;
```
Comment (RomanDubovskyi) [inline src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java]:
```suggestion
            }
            return manufacturers;
```
statement is inside of try with resources, we don't need to close it explicitly in try

File: src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+    private Manufacturer createManufacturer(ResultSet resultSet) {
+        try {
+            String name = resultSet.getString("manufacturer_name");
+            String country = resultSet.getString("manufacturer_country");
+            Long manufacturerId = resultSet.getObject("manufacturer_id", Long.class);
+            Manufacturer manufacturer = new Manufacturer(name, country);
+            manufacturer.setId(manufacturerId);
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create manufacturer", e);
```
Comment (RomanDubovskyi) [inline src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java]:
```suggestion
            throw new DataProcessingException("Can't parse manufacturer from result set ", e);
```

File: travis.yml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+language: java
+jdk:
+  - openjdk11
+script:
+  - mvn clean verify
+cache:
+  directories:
+    - $HOME/.m2
```
Comment (RomanDubovskyi) [inline travis.yml]: missing an empty line at the end of the file

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java (hunk 5 above)
Comment (Alex-Korzun) [reply src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java]: changed column name

File: src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java (hunk 6 above)
Comment (Alex-Korzun) [reply src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java]: sorry. my mistake. forgot to remove it after changing

File: src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java (hunk 7 above)
Comment (Alex-Korzun) [reply src/main/java/manufacturing/dao/jdbc/ManufacturerDaoJdbcImpl.java]: changed

File: travis.yml (hunk 8 above)
Comment (Alex-Korzun) [reply travis.yml]: done

File: travis.yml [hunk 9]
```diff
@@ -5,4 +5,5 @@ script:
   - mvn clean verify
 cache:
   directories:
-    - $HOME/.m2
\ No newline at end of file
+    - $HOME/.m2
+    -
```
Comment (RomanDubovskyi) [inline travis.yml]: you should add an empty line, not a line with "-" symbov

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**